compiler/rustc_ty_utils/src/needs_drop.rs

```diff
@@ -194,7 +194,11 @@ where
                             queue_type(self, required);
                         }
                     }
-                    ty::Alias(..) | ty::Array(..) | ty::Placeholder(_) | ty::Param(_) => {
+                    ty::Pat(..)
```

Comment:
I'm confused why `Pat` is being handled here. It shouldn't be needed -- we're already projecting it to its (inner_ty) component elsewhere.

Like, `Placeholder` and `Param` are needed b/c they're unsubstituted; `Array` is needed because we have the `0` case for drop. But `Pat` shouldn't be needed here.